sdcm/utils/version_utils.py

```diff
@@ -120,7 +120,9 @@ class ScyllaFileType(Enum):
 
 def parse_scylla_version(version_to_parse: str) -> version.Version:
     version_format = re.compile(r'(\d+\.\d)|(\.\d+)')
-    return parse_version(version_format.search(version_to_parse)[0])
+    major_version = version_format.search(version_to_parse)
+    assert major_version, f"version_to_parse: '{version_to_parse}' isn't supported scylla version string"
+    return parse_version(major_version[0])
```

Comment:
nitpick: 
`major_version` is only the 1st item of the tuple created by `version_format.search(version_to_parse)` IIRC
so it should be something like `(2022, 1, 0)`

Reply:
the logic is to extract the major version, the reason the first index is taken on line 125, is cause it can be None.

Reply:
if you think the version should include the minor as well, then this code was broken to begin with...

Reply:
not true, because `re.compile(r'(\d+\.\d)|(\.\d+)')` will return `x.y` and it is major and minor versions

so, when you have as a version `2021.1.8`, the result will be `major=2021`, `minor=1` and `micro=8`

we can consider as our major versions the whole `x.y`, but this is not the most correct

Reply:
that's not the terminology we use here in scylla for our versions:

2022.1 is a major version
2022.1.8 is a minor version

Reply:
> that's not the terminology we use here in scylla for our versions:
> 
> 2022.1 is a major version 2022.1.8 is a minor version

wrong, `2022` is major version, `1` is minor version, `8` is build/micro version

as we have now 2022.1 and 2022.2, although we may call them as "version" or "major version", from version convention, what you said isn't accurate